nrfe/nrfe_gpio.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2024, Nordic Semiconductor ASA
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef NRFE_GPIO_H
+#define NRFE_GPIO_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief eGPIO opcodes. */
+typedef enum
+{
+    NRFE_GPIO_PIN_CONFIGURE	= 0, ///< Configure eGPIO pin.
+    NRFE_GPIO_PIN_CLEAR    	= 1, ///< Clear eGPIO pin.
+    NRFE_GPIO_PIN_SET      	= 2, ///< Set eGPIO pin.
+    NRFE_GPIO_PIN_TOGGLE   	= 3, ///< Toggle eGPIO pin.
+} nrfe_gpio_opcode_t;
+
+/** @brief eGPIO data packet. */
+typedef struct {
+	nrfe_gpio_opcode_t  opcode;	/* eGPIO opcode. */
```

Comment:
I would prefer defines instead of `nrfe_gpio_opcode_t` enum. Then in the `nrfe_gpio_data_packet_t` structure you can use the uint8_t type, which is completely sufficient. This will additionally limit the generation of type warnings and standardize/set the size of the `opcode` variable.
Additionally, it would be good to make the `nrfe_gpio_data_packet_t` structure aligned to the 1 byte. This is good practice in communication structures to avoid errors related to different structure packaging for different architectures/compilation flags etc.

Reply:
> I would prefer defines instead of nrfe_gpio_opcode_t enum. Then in the nrfe_gpio_data_packet_t structure you can use the uint8_t type, which is completely sufficient. This will additionally limit the generation of type warnings and standardize/set the size of the opcode variable.

Right, I haven't thought about it. What if the opcode type was `uint8_t` and `nrfe_gpio_opcode_t` was cast to `uint8_t` when being assigned to opcode? I would want to avoid replacing an enum with defines as it might be unclear what the possible options are.

Reply:
Wouldn't it be enough to make this struct `packed`?
```
When attached to an enum definition, it indicates that the smallest integral type should be used.
Specifying this attribute for struct and union types is equivalent to specifying the packed attribute on each of the structure or union members.
```

Reply:
I was more concerned with the packing of structure than with a specific type. However, good practice is to make the type explicit. If we use type casting, there should be no type warnings in the case of an enum.

Reply:
Changed opcode to `uint8_t` type and struct definition to packed.